tools/migrator/src/utils/checkGitStatus.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-console */
+import chalk from 'chalk';
+import isGitClean from 'is-git-clean';
+
+/**
+ * Check whether a Git repository is in a "clean" state, meaning there are no uncommitted changes or untracked files in the repository
+ * @param force boolean. Whether to forcibly continue even though the git directory is not clean
+ */
+export function checkGitStatus(force?: boolean) {
+  let clean = false;
+  let errorMessage = 'Unable to determine if git directory is clean';
+
+  try {
+    clean = isGitClean.sync(process.cwd());
+    errorMessage = 'Git directory is not clean';
+  } catch (err: any) {
+    if (err && err.stderr && err.stderr.indexOf('Not a git repository') >= 0) {
+      clean = true;
+    }
+  }
+
+  if (!clean) {
+    if (force) {
```

Comment:
should we check for `force` before the `try/catch`?

Reply:
We can but then we won't know if the git directory is clean or not so we can't console log the explicit message, `Git directory is not clean`. 
Also, currently, if this isn't a git directory and the force flag is passed, we skip the force check, but if we check for force first we'll end up console logging `Forcibly continuing`